pkgs/os-specific/linux/musl/default.nix

```diff
@@ -37,6 +37,7 @@ let
       then "i386"
       else null;
 
+  enableShared = !(stdenv.hostPlatform.isx86_32 && stdenv.hostPlatform.useLLVM);
```

Comment:
Please justify this, why would i686 + LLVM need to disable shared libs?

Reply:
I am going to cc @Ericson2314 in here